### DISCUSSION TITLE
How to solve conflict between mui5 and mui4?

### DISCUSSION THREAD

**Discussion by RandyLiu6410** [react-page/react-page#1164] [Q&A]
My project has just been updated to Mui v5 (https://mui.com/). And some conflicts just showed up. When I use the module importing the react-page, the style sheet of mui v4 covered the current style sheet of mui v5. How can I prevent mui v4 from covering all style sheets and make sure mui v5 is still working as before?

**Comment by macrozone** [comment]
unfortunatly, MUI has a history in not playing well with multiple versions of material-ui, that's why its a peerDependency instead of a direct dependency (which would be much better!)

I did not update any project yet, react-page will definitly move to mui5 (or maybe something else that is not leaking its styles...).


i also saw your stack overflow: https://stackoverflow.com/questions/71254591/how-to-solve-conflict-between-my-project-using-mui5-and-package-using-mui4

maybe also post your questions here https://github.com/mui/material-ui/discussions

**Reply by macrozone** [reply]
having said that, multiple versions should work in theory: https://github.com/mui/material-ui/issues/17186

we also have actually a custom seed there in the classname generator

but i do not rule out a problem on our side, maybe we should take a closer look at that
